codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/eventstream/EventStreamUnmarshallerGeneratorTest.kt

```diff
@@ -0,0 +1,94 @@
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.core.smithy.generators.BuilderGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.generators.implBlock
+import software.amazon.smithy.rust.codegen.core.smithy.protocols.Protocol
+import software.amazon.smithy.rust.codegen.core.smithy.protocols.parse.EventStreamUnmarshallerGenerator
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestModels
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestTools
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestVariety
+import software.amazon.smithy.rust.codegen.core.testutil.TestEventStreamProject
+import software.amazon.smithy.rust.codegen.server.smithy.ServerCodegenContext
+import software.amazon.smithy.rust.codegen.server.smithy.generators.serverBuilderSymbol
+import java.util.stream.Stream
+
+data class TestCase(
+    val eventStreamTestCase: EventStreamTestModels.TestCase,
+    val publicConstrainedTypes: Boolean,
+) {
+    override fun toString(): String = "$eventStreamTestCase, publicConstrainedTypes = $publicConstrainedTypes"
+}
+
+class UnmarshallTestCasesProvider : ArgumentsProvider {
+    override fun provideArguments(context: ExtensionContext?): Stream<out Arguments> =
+        EventStreamTestModels.TEST_CASES.flatMap { testCase ->
+            listOf(
+                // TODO(https://github.com/awslabs/smithy-rs/issues/1442): Enable tests for `publicConstrainedTypes = false`
+                // TestCase(testCase, false),
+                TestCase(testCase, true),
+            )
+        }.map { Arguments.of(it) }.stream()
+}
+
+class EventStreamUnmarshallerGeneratorTest {
+    @ParameterizedTest
+    @ArgumentsSource(UnmarshallTestCasesProvider::class)
+    fun test(testCase: TestCase) {
+        EventStreamTestTools.runTestCase(
+            testCase.eventStreamTestCase,
+            object : EventStreamBaseRequirements() {
+                override fun renderGenerator(
+                    codegenContext: ServerCodegenContext,
+                    project: TestEventStreamProject,
+                    protocol: Protocol,
+                ): RuntimeType {
+                    fun builderSymbol(shape: StructureShape): Symbol = shape.serverBuilderSymbol(codegenContext)
+                    return EventStreamUnmarshallerGenerator(
+                        protocol,
+                        codegenContext,
+                        project.operationShape,
+                        project.streamShape,
+                        ::builderSymbol,
+                    ).render()
+                }
+
+                override fun renderBuilderForShape(
+                    writer: RustWriter,
+                    codegenContext: ServerCodegenContext,
+                    shape: StructureShape,
+                ) {
+                    // TODO(https://github.com/awslabs/smithy-rs/issues/1442): Use the correct builder:
```

Comment:
If you're punting on this, mention in the TODO that we need to test with two builders: `ServerBuilderGeneratorWithoutPublicConstrainedTypes` and `ServerBuilderGenerator`.

Same goes for the `software.amazon.smithy.rust.codegen.server.smithy.protocols.eventstream.EventStreamMarshallerGeneratorTest` class, which is only testing with `ServerBuilderGenerator`.

Reply:
Fixed in a47597fccc413b489e6c9d3f93dc3c2149a2036c.

Reply:
Fixed the marshaller side of things in 0b77d8ffe7380d07a0ff648ae4e526ecc28616d7.